packages/astro/src/runtime/server/render/astro/factory.ts

```diff
@@ -26,3 +26,14 @@ export function isAPropagatingComponent(
 	}
 	return hint === 'in-tree' || hint === 'self';
 }
+
+export function getPropagationHint(
```

Comment:
don't we have this same function somewhere else? Or is it different? If the same we should reuse.

Reply:
The other function is in the same file, few lines up. It's different and it checks if `hint === 'in-tree' || hint === 'self'`.

Reply:
Shouldn't that function use this one then?

Reply:
Makes sense!